android/src/main/java/com/reactnativeone/OneModule.java

```diff
@@ -72,62 +82,160 @@ public void init(String siteKey, String touchpointUri, String apiKey,
+          notifyResult(promise, responseMap);
+        } catch (ExecutionException error) {
+          notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Interaction Completion Error: " + error.getCause());
+        } catch (OneSDKError error) {
+          notifyProblem(promise, Integer.toString(error.getSystemCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Interaction SDK Error: " + error.getErrorMessage());
+        } catch (OneAPIError error) {
+          notifyProblem(promise, Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Interaction Api Error: " + error.getErrorMessage());
+        } catch (Exception error) {
+          notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Interaction Error: " + error.getLocalizedMessage());
         }
-      });
-    }
+      }
+      WritableNativeMap responseMap = responseObjectToReadableMap(response);
+      notifyResult(promise, responseMap);
+    });
   }
 
   @ReactMethod
-  public void sendProperties(String interaction, ReadableMap propertiesMap) {
+  public void sendProperties(String interactionPath, ReadableMap propertiesMap, final Promise promise) {
     HashMap<String, String> properties = getPropertiesFromReadableMap(propertiesMap);
     final OneRequest sendPropertiesRequest = new OneRequest.Builder()
-      .interactionPath(new OneInteractionPath(URI.create(interaction)))
+      .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
       .properties(properties)
       .build();
-
-    final OneCall sendPropertiesCall = One.sendProperties(sendPropertiesRequest);
-    sendPropertiesCall.enqueue(null);
+    executor.submit(() -> {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        try {
+          OneResponse response;
+          response = One.sendProperties(THROW_ERRORS, sendPropertiesRequest).join();
+          WritableNativeMap responseMap = responseObjectToReadableMap(response);
+          notifyResult(promise, responseMap);
+        } catch (ExecutionException error) {
+          notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Properties Completion Error: " + error.getCause());
+        } catch (OneSDKError error) {
+          notifyProblem(promise, Integer.toString(error.getSystemCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Properties SDK Error: " + error.getErrorMessage());
+        } catch (OneAPIError error) {
+          notifyProblem(promise, Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Properties Api Error: " + error.getErrorMessage());
+        } catch (CompletionException error) {
```

Comment:
You should return the cause  message for better insight.

Reply:
are you suggesting returning the `cause` message in the Log only, the Promise, or both?

Reply:
updated, fyi the cause message required `Objects.requireNonNull` around it.  

i.e. `Objects.requireNonNull(error.getCause()).getLocalizedMessage()`